src/ray/core_worker/core_worker.cc

```diff
@@ -592,13 +713,13 @@ CoreWorker::CoreWorker(const CoreWorkerOptions &options, const WorkerID &worker_
       for (const auto &node_id : object_locations.value()) {
         absl::optional<rpc::Address> addr = node_addr_factory(node_id);
         if (addr.has_value()) {
-          locations.push_back(addr.value());
-        } else {
-          // We're getting potentially stale locations directly from the reference
-          // counter, so the location might be a dead node.
-          RAY_LOG(DEBUG).WithField(object_id).WithField(node_id)
-              << "Object location is dead, not using it in the recovery of object";
+          locations.emplace_back(std::move(addr.value()));
+          continue;
```

Comment:
why do we do this `continue`? I find this makes the code control flow harder to follow.

Reply:
clang tidy has an item for early return: https://clang.llvm.org/extra/clang-tidy/checks/readability/else-after-return.html
The motivation of which is to reduce indentation.

Reply:
Well the code block is <10 LOC..
```
for (const auto &node_id : object_locations.value()) {
        absl::optional<rpc::Address> addr = node_addr_factory(node_id);
        if (addr.has_value()) {
          locations.emplace_back(std::move(addr.value()));
          continue;
        }
        // We're getting potentially stale locations directly from the reference
        // counter, so the location might be a dead node.
        RAY_LOG(DEBUG).WithField(object_id).WithField(node_id)
            << "Object location is dead, not using it in the recovery of object";
      }
```

Reply:
If you really want, I would revert the change, but overall early exit actually helps readability, especially when you're in deep nested indentation.

Reply:
I'd argue the "early exit helps readability" only applies for a function (or lambda) but not to a `continue` in a for loop. The if-else ensures "for each element, it runs exactly either `then` or `else` once, and the reasoning is easier for me.

But I don't have a strong stance on this either. We can live with 2 styles in codebase.